src/graphql/exprParser.js

```diff
@@ -0,0 +1,423 @@
+            unit = 's';
+        }
+        else if (ExprParser.isKeywordMin(currChar)) {
+            unit = 'm';
+        }
+        else if (ExprParser.isKeywordHour(currChar)) {
+            unit = 'h';
+        }
+        else if (ExprParser.isKeywordDay(currChar)) {
+            unit = 'd';
+        }
+        else if (ExprParser.isKeywordMonth(currChar)) {
+            unit = 'M';
+        }
+        else {
+            // None of the above
+            return unit;
+        }
+
+        ExprParser.consumeChar(pos);
+        return unit;
+    }
+
+    // returns a string of numbers
+    static consumeNumbers(charArr, pos) {
+        let numberStr = "";
+        let consumed = 0;
+
+        for (let index = pos.current;
+            index < charArr.length;
+            index ++) {
+            let currChar = charArr[index];
+
+            if (!isNaN(currChar)) {
+                numberStr = numberStr + currChar;
+                consumed += 1;
+            }
+            else {
+                // No more numbers
+                break;
+            }
+        }
+
+        pos.current += consumed;
+        return numberStr;
+    }
+
+    static consumeChar(position) {
+        position.current += 1;
+    }
+
+    static consumeWhitespaces(charArr, pos) {
+        let consumed = 0;
+
+        for (let index = pos.current;
+            index < charArr.length;
+            index ++) {
+            let currChar = charArr[index];
+
+            if (currChar === WHITESPACE) {
+                consumed += 1;
+            }
+            else {
+                // No more whitespace
+                break;
+            }
+        }
+
+        pos.current += consumed;
+    }
+
+    static isKeywordDay(char) {
+        return char === 'd';
+    }
+
+    static isKeywordSec(char) {
+        return char === 's';
+    }
+
+    static isKeywordMin(char) {
+        return char === 'm';
+    }
+
+    static isKeywordHour(char) {
+        return char === 'h';
+    }
+
+    static isKeywordMonth(char) {
+        return char === 'M';
+    }
+
+    static isKeywordSuffixInd(charArr, pos) {
+        let isSuffixInd = false;
+
+        if (charArr[pos.current] === '/') {
+            isSuffixInd = true;
+        }
+
+        return isSuffixInd;
+    }
+
+    static isKeywordAdd(charArr, pos) {
+        let isAddition = false;
+
+        if (charArr[pos.current] === '+') {
+            isAddition = true;
+        }
+
+        return isAddition;
+    }
+
+    static isKeywordSub(charArr, pos) {
+        let isSubtract = false;
+
+        if (charArr[pos.current] === '-') {
+            isSubtract = true;
+        }
+
+        return isSubtract;
+    }
+
+    static isKeywordNow(charArr, pos) {
+        let isNow = false;
+
+        if (charArr[pos.current].toLowerCase() === 'n' &&
+            charArr[pos.current + 1].toLowerCase() === 'o' &&
+            charArr[pos.current + 2].toLowerCase() === 'w') {
+            isNow = true;
+        }
```

Comment:
I think that `) {` is better on its own line, as the condition expression looks like it belongs with the `isNow = true;` due to ambiguous indentation.